packages/medusa/src/api/routes/store/variants/list-variants.ts

```diff
@@ -71,10 +71,35 @@ export default async (req, res) => {
     filterableFields.id = validated.ids.split(",")
   }
 
+  const pricingService: PricingService = req.scope.resolve("pricingService")
   const variantService: ProductVariantService = req.scope.resolve(
     "productVariantService"
   )
-  const variants = await variantService.list(filterableFields, listConfig)
+  const cartService: CartService = req.scope.resolve("cartService")
```

Comment:
issue: All operations should be wrapped into a transaction
suggestion: I would suggest to update the handler as follow:

```ts
const customer_id = req.user?.customer_id

  const pricingService: PricingService = req.scope.resolve("pricingService")
  const variantService: ProductVariantService = req.scope.resolve(
    "productVariantService"
  )
  const cartService: CartService = req.scope.resolve("cartService")
  const regionService: RegionService = req.scope.resolve("regionService")
  const manager: EntityManager = req.scope.resolve("manager")

  let {
    expand,
    offset,
    limit,
    ids,
    cart_id,
    region_id,
    currency_code,
    ...filterableFields
  } = await validator(StoreGetVariantsParams, req.query)

  let expandFields: string[] = []
  if (expand) {
    expandFields = expand.split(",").map((v) => v.trim())
  }

  const listConfig = {
    relations: expandFields.length
      ? expandFields
      : defaultStoreVariantRelations,
    skip: offset,
    take: limit,
  }

  if (ids) {
    filterableFields.id = ids.split(",").map((v) => v.trim())
  }

  const variants = await manager.transaction(
    async (transactionManager: EntityManager) => {
      const rawVariants = await variantService
        .withTransaction(transactionManager)
        .list(filterableFields, listConfig)

      if (cart_id) {
        const cart = await cartService
          .withTransaction(transactionManager)
          .retrieve(cart_id, {
            select: ["id", "region_id"],
          })
        const region = await regionService
          .withTransaction(transactionManager)
          .retrieve(cart.region_id, {
            select: ["id", "currency_code"],
          })
        region_id = region.id
        currency_code = region.currency_code
      }

      return await pricingService.setVariantPrices(rawVariants, {
        cart_id,
        region_id,
        currency_code,
        customer_id,
        include_discount_prices: true,
      })
    }
  )

  res.json({ variants })
```

wdyt?